app/services/session.js

```diff
@@ -34,26 +36,30 @@ export default Session.extend({
   /**
    * Get the account information for the sessioned user
    */
-  getCurrentUser() {
-    const requestUrl = '/users?filter[self]=true&include=userRoles.role,userRoles.user';
-    return get(this, 'ajax').request(requestUrl).then((response) => {
-      // push the user data into the store
-      const [data] = response.data;
-      const normalizedData = get(this, 'store').normalize('user', data);
-      const user = get(this, 'store').push(normalizedData);
-      // push any included data into the store
-      const included = response.included || [];
-      included.forEach((record) => {
-        let type = get(record, 'type');
-        type = type === 'userRoles' ? 'user-role' : 'role';
-        get(this, 'store').push(get(this, 'store').normalize(type, record));
-      });
-      // set reference of the user
-      set(this, 'account', user);
+  async getCurrentUser() {
+    const { store, raven } = getProperties(this, 'store', 'raven');
+    try {
+      // Load the current user
+      const user = get(await store.query('user', {
+        filter: { self: true },
+        include: 'userRoles.role,userRoles.user'
+      }), 'firstObject');
```

Comment:
Could we extract this to 2 variables? Easier to read and then update when we can use ES5 getters